### PR TITLE
add transformState solution

### PR DESCRIPTION
**Pull request by booobryyyk**
add transformState solution

### CODE REVIEW

File: src/transformState.js [hunk 1]
```diff
@@ -5,7 +5,30 @@
...
+      case 'addProperties':
+        for (const data in action.extraData) {
+          state[data] = action.extraData[data];
+        }
+        break;
+
+      case 'clear':
+        for (const key in state) {
+          delete state[key];
+        }
```
Comment (dinaminahi) [inline src/transformState.js]: Let's create a separate function for this to make `switch-case` smaller and more readable

File: src/transformState.js [hunk 2]
```diff
@@ -5,7 +5,30 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        for (const data in action.extraData) {
+          state[data] = action.extraData[data];
+        }
```
Comment (dinaminahi) [inline src/transformState.js]: It`s better to use Object.assign or spread operator here

File: src/transformState.js [hunk 3]
```diff
@@ -5,7 +5,30 @@
...
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete state[key];
+        }
+        break;
+
+      default:
+        throw new Error(`There is no such action type: ${action.type}`);
```
Comment (dinaminahi) [inline src/transformState.js:26]: I like that you throw an error with a clear description 👍

**Review by dinaminahi — CHANGES_REQUESTED** [review]

File: src/transformState.js [hunk 4]
```diff
@@ -5,7 +5,34 @@
...
+
+      case 'clear':
+        clearObject(state);
+
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete state[key];
+        }
```
Comment (33Orange) [inline src/transformState.js]: Let's put this in a separate function, just like `clearObject`

**Review by 33Orange — CHANGES_REQUESTED** [review]
Well done! 🔥 

Just one moment with readability

**Review by Anna-Tkachenko — APPROVED** [review]
⭐️

**Pull request closed without merging**